src/core/org/apache/jmeter/gui/action/HtmlReportGenerator.java

```diff
@@ -0,0 +1,186 @@
+        List<String> arguments = new ArrayList<>();
+        String java = System.getProperty("java.home") + "/bin/java";
+        arguments.add(java);
+        arguments.add("-jar");
+        arguments.add(jmeterBinDir + "/ApacheJMeter.jar");
+        arguments.add("-p");
+        arguments.add(jmeterBinDir + "/jmeter.properties");
+        arguments.add("-q");
+        arguments.add(userPropertiesFilePath);
+        arguments.add("-g");
+        arguments.add(csvFilePath);
+        arguments.add("-j");
+        arguments.add(jmeterBinDir + "/jmeter_html_report.log");
+        arguments.add("-o");
+        arguments.add(outputDirectoryPath);
+        return arguments;
+    }
+
+    /**
+     * test that all arguments are correct and send a message to the user if not
+     * 
+     * @return whether or not the files are correct
+     */
+    private List<String> checkArguments() {
+        List<String> errors = new ArrayList<>();
+
+        String csvError = checkFile(new File(csvFilePath));
+        if (csvError != null) {
+            errors.add(JMeterUtils.getResString("generate_report_ui.csv_file") + csvError);
+        }
+
+        String userPropertiesError = checkFile(new File(userPropertiesFilePath));
+        if (userPropertiesError != null) {
+            errors.add(JMeterUtils.getResString("generate_report_ui.user_properties_file") + userPropertiesError);
+        }
+
+        String outputError = checkDirectory(new File(outputDirectoryPath));
+        if (outputError != null) {
+            errors.add(JMeterUtils.getResString("generate_report_ui.output_directory") + outputError);
+        }
+        return errors;
+    }
+
+    /**
+     * Check if a file is correct for report generation
+     * 
+     * @param fileToCheck
+     *            the directory to check
+     * @return the error message or null if the file is ok
+     */
+    private String checkFile(File fileToCheck) {
+        if (fileToCheck.exists() && fileToCheck.canRead() && fileToCheck.isFile()) {
+            return null;
+        } else {
+            return MessageFormat.format(JMeterUtils.getResString(NO_FILE), fileToCheck);
+        }
+    }
+
+    /**
+     * Check if a directory is fine for report generation
+     * 
+     * @param directoryToCheck
+     *            the directory to check
+     * @return the error message or an empty string if the directory is fine
+     */
+    private String checkDirectory(File directoryToCheck) {
+        if (directoryToCheck.exists()) {
+            String[] files = directoryToCheck.list();
+            if (files != null && files.length > 0) {
+                return MessageFormat.format(JMeterUtils.getResString(NOT_EMPTY_DIRECTORY), directoryToCheck);
+            } else {
+                return null;
+            }
+        } else {
+            File parentDirectory = directoryToCheck.getParentFile();
+            if(parentDirectory != null && parentDirectory.exists() && parentDirectory.canWrite()) {
```

Comment:
Why did not you just use `directoryToCheck.mkdirs()`?

Reply:
By convention for now,  we only create one level of directory. Mkdirs would create the whole hierarchy